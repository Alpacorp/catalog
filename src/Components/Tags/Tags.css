.product-info .tags small {
  background-color: blueviolet;
  border-radius: 5px;
  color: #fff;
  font-size: 10px;
  margin-block-start: 3px;
  margin-inline-end: 5px;
  padding: 5px;
  text-transform: capitalize;
}
