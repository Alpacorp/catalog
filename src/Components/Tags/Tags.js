import PropTypes from "prop-types";
import "./Tags.css";

const Tags = ({ tags, id }) => {
  return (
    <div className="tags" tabIndex="0">
      {tags?.map((tag) => (
        <small aria-label={tag} className="tag" key={id + tag}>
          <strong>{tag}</strong>
        </small>
      ))}
    </div>
  );
};

Tags.propTypes = {
  tags: PropTypes.array.isRequired,
  id: PropTypes.number.isRequired,
};

export default Tags;
