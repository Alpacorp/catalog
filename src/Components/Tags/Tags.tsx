import "./Tags.css";

interface TagsProps {
  tags: string[];
  id: number;
}

const Tags = ({ tags, id }: TagsProps) => {
  return (
    <div data-testid="test-div-tags" className="tags" tabIndex={0}>
      {tags?.map((tag) => (
        <small
          data-testid="test-small-tag"
          aria-label={tag}
          className="tag"
          key={id + tag}
        >
          <strong>{tag}</strong>
        </small>
      ))}
    </div>
  );
};

export default Tags;
