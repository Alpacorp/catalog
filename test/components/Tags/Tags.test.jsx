import { render, screen } from "@testing-library/react";
import Tags from "../../../src/Components/Tags/Tags";
import propTypes from "prop-types";

describe("Test on <Tags /> component", () => {
  const tags = ["tag1"];
  const id = 1;

  test("Component should render to match snapshot", () => {
    render(<Tags tags={tags} id={id} />);
    expect(screen.getByTestId("test-div-tags")).toMatchSnapshot();
  });

  test("The component have two props is requiered", () => {
    render(<Tags tags={tags} id={id} />);
    expect(Tags.propTypes).toEqual({
      tags: propTypes.array.isRequired,
      id: propTypes.number.isRequired,
    });
    expect(Object.keys(Tags.propTypes).length).toBe(2);
  });

  test('The component main div to have "tags" className and <small> tag to have "tag" className', () => {
    render(<Tags tags={tags} id={id} />);
    expect(screen.getByTestId("test-div-tags").className).toBe("tags");
    expect(screen.getByTestId("test-small-tag").className).toBe("tag");
  });
});
