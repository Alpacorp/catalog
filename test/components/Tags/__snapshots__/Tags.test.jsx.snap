// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <Tags /> component Component should render to match snapshot 1`] = `
<div
  class="tags"
  data-testid="test-div-tags"
  tabindex="0"
>
  <small
    aria-label="tag1"
    class="tag"
    data-testid="test-small-tag"
  >
    <strong>
      tag1
    </strong>
  </small>
</div>
`;
